CMSgov/mint-app
# Easi 2434/mac and init user table

# EASI-2434

## Changes and Description

- Introduce a User Table
    -  Added to the user table at request initialization
    - User Info retrieved from OktaEndpoint and stored in the DB
- Refactor NDA Agreement
   - Table now references the user table instead of EUAID


## How to test this change

1. Log into the App using your OKTA EUID Account
2.  Confirm you can log in and accept the NDA. 
3. Confirm there is a record of your account in the user table
4. If you have a test account with MAC user job codes, log in and repeat the steps for MAC


## Future Improvements
- Eventually, every table can be refactored to use the user table instead of using EUAID
- The user info endpoint is currently being constructed by combining a root URL with `/userinfo`, the actual endpoint is exposed in the frontend token, but not the backend token. It would be more ideal to get the full endpoint from there.
- The test suite should eventually add a user record to the table before every test script. I skipped that for now as it is unneeded. (This is because the account is created before a graphql request is handled.

## Front End Considerations
- With this, the MAC user functionality is mostly there, but the ability to create a new model plan needs to be removed. That will currently error out on the front end.

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
